spec/unit/pops/evaluator/evaluating_parser_spec.rb

```diff
@@ -1040,6 +1040,12 @@ def assert_no_undef(x)
       expect{parser.evaluate_string(scope, "assert_no_undef({1 => undef})")}.to_not raise_error()
     end
 
+    it 'a lambda return value is checked using the return type' do
+      expect(parser.evaluate_string(scope, "[1,2].map |Integer $x| >> Integer { $x }")).to eql([1,2])
+      expect { parser.evaluate_string(scope, "[1,2].map |Integer $x| >> String { $x }") }.to raise_error(
+        /value returned from lambda had wrong type, expected a String value, got Integer/)
+    end
+
```

Comment:
It doesn't check against Any when not specified. It doesn't check at all. It _could_ check against Any of course, and that would have been just as correct but whether or not it does is an implementation detail and I see no value in writing a unit test with an expectation for a certain type of check (or lack thereof).

I'm not comfortable writing a test using some arbitrary types either. There are already a lot of tests that calls methods returning all kinds of objects (all function calls that take in the test framework). If some unexpected test using some unexpected type would take place when a function returned, then hundreds of tests would break. So what would be the point in adding yet another test?